addons/opacity-slider/userscript.js

```diff
@@ -0,0 +1,212 @@
+    //   state.scratchPaint.selectedItems[i].fillColor.set(color);
+    // }
+    // }, 500);
+  };
+
+  const setSliderBg = (color) => {
+    const hex = tinycolor(color).toHexString(); // remove alpha value
+    const darkmodeBackground = getComputedStyle(document.documentElement).getPropertyValue(
+      "--editorDarkMode-accent-paintEditorBackground"
+    );
+    const bgColor = darkmodeBackground === "" ? "#eaf0f8" : darkmodeBackground;
+    let bg = `linear-gradient(to left, ${hex} 0%, rgba(0, 0, 0, 0) 100%),`;
+    bg += `linear-gradient(45deg, ${bgColor} 25%, transparent 25%, transparent 75%, ${bgColor} 75%),`;
+    bg += `linear-gradient(45deg, ${bgColor} 25%, transparent 25%, transparent 75%, ${bgColor} 75%)`;
+    saOpacitySlider.style.background = bg;
+    saOpacitySlider.style.backgroundSize = "100% 100%, 20px 20px, 20px 20px";
+    saOpacitySlider.style.backgroundPosition = "0 0, 10px 10px";
+  };
+
+  const getEventXY = (e) => {
+    if (e.touches && e.touches[0]) {
+      return { x: e.touches[0].clientX, y: e.touches[0].clientY };
+    } else if (e.changedTouches && e.changedTouches[0]) {
+      return { x: e.changedTouches[0].clientX, y: e.changedTouches[0].clientY };
+    }
+    return { x: e.clientX, y: e.clientY };
+  };
+
+  const handleMouseDown = (event) => {
+    document.addEventListener("mousemove", handleMouseMove);
+    document.addEventListener("mouseup", handleMouseUp);
+    ignoreKeepingOpacity = true;
+    handleClickOffset = getEventXY(event).x - saOpacityHandle.getBoundingClientRect().left;
+  };
+
+  const handleMouseUp = () => {
+    document.removeEventListener("mousemove", handleMouseMove);
+    document.removeEventListener("mouseup", handleMouseUp);
+    ignoreKeepingOpacity = false;
+  };
+
+  const handleMouseMove = (event) => {
+    event.preventDefault();
+    ignoreKeepingOpacity = true;
+    changeOpacity(scaleMouseToSliderPosition(event));
+  };
+
+  const handleClickBackground = (event) => {
+    if (event.target !== saOpacitySlider) return;
+    handleClickOffset = HANDLE_WIDTH / 2;
+    ignoreKeepingOpacity = true;
+    changeOpacity(scaleMouseToSliderPosition(event));
+  };
+
+  const scaleMouseToSliderPosition = (event) => {
+    const { x } = getEventXY(event);
+    const backgroundBBox = saOpacitySlider.getBoundingClientRect();
+    const scaledX = x - backgroundBBox.left - handleClickOffset;
+    return Math.max(0, Math.min(100, (100 * scaledX) / (backgroundBBox.width - HANDLE_WIDTH)));
+  };
+
+  const changeOpacity = (opacityValue) => {
+    const halfHandleWidth = HANDLE_WIDTH / 2;
+    const pixelMin = halfHandleWidth;
+    const pixelMax = CONTAINER_WIDTH - halfHandleWidth;
+    labelReadout.textContent = Math.round(opacityValue);
+    saOpacityHandle.style.left = pixelMin + (pixelMax - pixelMin) * (opacityValue / 100) - halfHandleWidth + "px";
+
+    const color = tinycolor(getColor()).toRgb();
+    window.opacitySliderAlpha = opacityValue / 100;
```

Comment:
Could we use `scratchAddons.opacitySliderAlpha` instead of `window.opacitySliderAlpha`?
Userscripts don't often add new properties to the window object